library/core/src/main/java/com/google/android/exoplayer2/text/cea/Cea608Decoder.java

```diff
@@ -284,6 +328,24 @@ protected void decode(SubtitleInputBuffer inputBuffer) {
       // If we've reached this point then there is data to process; flag that work has been done.
       captionDataProcessed = true;
 
+      if ((ccDataHeader & CC_VALID_FLAG) != CC_VALID_FLAG) {
+        // The screen needs to be cleared when the closed captions are turned off
+        // (the encoder flips the validity bit)
+        resetCueBuilders();
+        continue;
+      }
+
+      // When transmission is digital and reliable (TCP) we should not meet any data corruption.
+      // But any over-the-air content and test streams might give the decoder corrupted data, so we
+      // need to drop those. When corrupt data is met the screen needs to be cleared, so we:
+      // - reset the cue builders
+      // - update the screen (by having this check after having captionDataProcessed set above)
+      if (byte1ParityBit == ODD_PARITY_BYTE_TABLE[ccData1]
```

Comment:
I think it's equivalent and simpler to do:
```
if (!ODD_PARITY_BYTE_TABLE[ccFullByte1] || !ODD_PARITY_BYTE_TABLE[ccFullByte2])
```
or better:
```
if (EVEN_PARITY_BYTE_TABLE[ccFullByte1] || EVEN_PARITY_BYTE_TABLE[ccFullByte2])
```
and then get rid of `byte1ParityBit` and `byte2ParityBit`?

Reply:
good idea!

Reply:
Spec describes all valid bytes having odd parity. So I kept that version and added a comment about the requirement.